src/test/java/com/google/jenkins/plugins/k8sengine/client/ContainerClientTest.java

```diff
@@ -16,108 +16,206 @@
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
+import static org.mockito.ArgumentMatchers.anyString;
 
 import com.google.api.services.container.Container;
+import com.google.api.services.container.Container.Projects.Zones.Clusters;
 import com.google.api.services.container.model.Cluster;
+import com.google.api.services.container.model.ListClustersResponse;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.Matchers;
-import org.mockito.Mock;
 import org.mockito.Mockito;
-import org.mockito.invocation.InvocationOnMock;
 import org.mockito.junit.MockitoJUnitRunner;
 import org.mockito.stubbing.Answer;
 
 /** Tests {@link ContainerClient}. */
 @RunWith(MockitoJUnitRunner.class)
 public class ContainerClientTest {
-  private ContainerClient containerClient;
-  @Mock private Container.Projects.Zones.Clusters clusters;
+  private static final String TEST_PROJECT_ID = "test-project-id";
+  private static final String OTHER_PROJECT_ID = "other-project-id";
+  private static final String ERROR_PROJECT_ID = "error-project-id";
+  private static final String EMPTY_PROJECT_ID = "empty-project-id";
+  private static final String TEST_ZONE = "us-west1-a";
+  private static final String OTHER_ZONE = "us-west2-b";
+  private static final String TEST_CLUSTER = "testCluster";
+  private static final String OTHER_CLUSTER = "otherCluster";
 
-  @Before
-  public void init() throws Exception {
+  private static ContainerClient containerClient;
+  private static Cluster testCluster;
+  private static List<Cluster> listOfClusters;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    listOfClusters = new ArrayList<>();
```

Comment:
Recommend avoiding introducing more dependencies on shared state like this. Admittedly there's already a bit of that with the shared baseline mocks, but anything we can do to lessen the re-factoring work for us going forward the better.